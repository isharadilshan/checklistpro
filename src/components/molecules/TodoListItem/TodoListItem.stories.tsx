import React from 'react';
import {NativeBaseProvider} from 'native-base';
import {ComponentMeta, ComponentStory} from '@storybook/react';

import TodoListItem from './';

export default {
  title: 'components/molecules/TodoListItem',
  component: TodoListItem,
} as ComponentMeta<typeof TodoListItem>;

export const Basic: ComponentStory<typeof TodoListItem> = () => {
  const args = {
    id: 'TODO_ID',
    title: 'Toto Title',
    description: 'Todo Description',
    status: 'INPROGRESS',
    category: 'MEDICAL',
    createdDate: 1680101338199,
    updatedDate: 1680101338199,
    hideButtons: false,
    onPressEdit: () => {},
    onPressDelete: () => {},
  };
  return (
    <NativeBaseProvider>
      <TodoListItem {...args} />
    </NativeBaseProvider>
  );
};
