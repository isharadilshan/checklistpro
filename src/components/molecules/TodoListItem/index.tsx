import React from 'react';
import {Box, Badge, HStack, View, Text} from 'native-base';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
import {getDateName} from '../../../utils/helper/Date';
import createStyle from './styles';

type TodoListItemProps = {
  id: string;
  title: string;
  description: string;
  status: string;
  category: string;
  createdDate: number;
  updatedDate: number;
  hideButtons: boolean;
  onPressEdit: ({}) => void;
  onPressDelete: ({}) => void;
};

const TodoListItem: React.FC<TodoListItemProps> = ({
  id,
  title,
  description,
  status,
  category,
  createdDate,
  updatedDate,
  hideButtons,
  onPressEdit,
  onPressDelete,
}) => {
  const styles = createStyle();

  const renderBadge = (colorScheme: string, status: string) => {
    return (
      <Badge width={100} variant={'subtle'} colorScheme={colorScheme}>
        {status}
      </Badge>
    );
  };

  const renderStatusBadge = (todoStatus) => {
    switch (todoStatus) {
      case 'TODO':
        return renderBadge('coolGray', todoStatus);
      case 'INPROGRESS':
        return renderBadge('warning', todoStatus);
      case 'HOLD':
        return renderBadge('error', todoStatus);
      case 'DONE':
        return renderBadge('info', todoStatus);
      default:
        return null;
    }
  };

  return (
    <Box style={styles.cardWrapper}>
      <HStack height={'100%'}>
        <Box w={'75%'}>
          <View style={styles.textWrapper}>
            <View>
              <Text color={'blueGray.400'} fontSize="xl">
                {title}
              </Text>
              <Text color={'blueGray.400'}>{description}</Text>
              <Text mt={3} color={'blueGray.400'}>
                {getDateName(updatedDate)}
              </Text>
            </View>
            <View style={styles.tagWrapper}>
              <Badge width={120} variant={'outline'} colorScheme="success">
                {category}
              </Badge>
              {renderStatusBadge(status)}
            </View>
          </View>
        </Box>
        {!hideButtons && (
          <Box w={'25%'}>
            <View style={styles.buttonWrapper}>
              <Icon
                name="square-edit-outline"
                size={30}
                style={styles.iconStyle}
                onPress={() =>
                  onPressEdit({
                    id,
                    title,
                    description,
                    status,
                    category,
                    createdDate,
                    updatedDate,
                  })
                }
              />
              <Icon
                name="delete"
                size={30}
                style={styles.iconStyle}
                onPress={() => onPressDelete({id})}
              />
            </View>
          </Box>
        )}
      </HStack>
    </Box>
  );
};

export default TodoListItem;
