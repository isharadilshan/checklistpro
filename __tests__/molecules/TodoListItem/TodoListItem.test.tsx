import React from 'react';
import {describe, expect} from '@jest/globals';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import TodoListItem from '../../../src/components/molecules/TodoListItem';

configure({adapter: new Adapter()});

const props = {
  id: 'TODO_ID',
  title: 'Toto Title',
  description: 'Todo Description',
  status: 'INPROGRESS',
  category: 'MEDICAL',
  createdDate: 1680101338199,
  updatedDate: 1680101338199,
  hideButtons: false,
  onPressEdit: jest.fn(),
  onPressDelete: jest.fn(),
};

describe('when rendering', () => {
  const wrapper = shallow(<TodoListItem {...props} />);

  it('should render a component matching the snapshot', () => {
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
    expect(wrapper).toHaveLength(1);
  });
});
