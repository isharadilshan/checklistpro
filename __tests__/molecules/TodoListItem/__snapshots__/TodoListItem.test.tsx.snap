// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when rendering should render a component matching the snapshot 1`] = `
<Memo(ForwardRef(Box))
  style={
    {
      "backgroundColor": "#1f2937",
      "borderColor": "#e5e7eb",
      "borderRadius": 4,
      "borderWidth": 1,
      "height": 150,
      "marginVertical": 10,
      "overflow": "hidden",
      "padding": 10,
    }
  }
>
  <Memo(ForwardRef(HStack))
    height="100%"
  >
    <Memo(ForwardRef(Box))
      w="75%"
    >
      <ForwardRef
        style={
          {
            "flex": 1,
            "justifyContent": "space-between",
          }
        }
      >
        <ForwardRef>
          <Memo(ForwardRef(Text))
            color="blueGray.400"
            fontSize="xl"
          >
            Toto Title
          </Memo(ForwardRef(Text))>
          <Memo(ForwardRef(Text))
            color="blueGray.400"
          >
            Todo Description
          </Memo(ForwardRef(Text))>
          <Memo(ForwardRef(Text))
            color="blueGray.400"
            mt={3}
          >
            Wednesday, March 29, 2023 8:18 PM
          </Memo(ForwardRef(Text))>
        </ForwardRef>
        <ForwardRef
          style={
            {
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Memo(ForwardRef(Badge))
            colorScheme="success"
            variant="outline"
            width={120}
          >
            MEDICAL
          </Memo(ForwardRef(Badge))>
          <Memo(ForwardRef(Badge))
            colorScheme="warning"
            variant="subtle"
            width={100}
          >
            INPROGRESS
          </Memo(ForwardRef(Badge))>
        </ForwardRef>
      </ForwardRef>
    </Memo(ForwardRef(Box))>
    <Memo(ForwardRef(Box))
      w="25%"
    >
      <ForwardRef
        style={
          {
            "alignItems": "flex-end",
            "flex": 1,
            "justifyContent": "space-between",
          }
        }
      >
        <Icon
          allowFontScaling={false}
          name="square-edit-outline"
          onPress={[Function]}
          size={30}
          style={
            {
              "color": "#94a3b8",
            }
          }
        />
        <Icon
          allowFontScaling={false}
          name="delete"
          onPress={[Function]}
          size={30}
          style={
            {
              "color": "#94a3b8",
            }
          }
        />
      </ForwardRef>
    </Memo(ForwardRef(Box))>
  </Memo(ForwardRef(HStack))>
</Memo(ForwardRef(Box))>
`;
